pkgs/development/python-modules/timing-asgi/default.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, poetry-core
+, asynctest
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "timing-asgi";
+  version = "0.3.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "steinnes";
+    repo = "timing-asgi";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-oEDesmy9t2m51Zd6Zg87qoYbfbDnejfrbjyBkZ3hF58=";
+  };
+
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+    --replace-fail poetry.masonry.api poetry.core.masonry.api \
+    --replace-fail "poetry>=" "poetry-core>="
+  '';
+
+  build-system = [
+    poetry-core
+  ];
+
+  pythonImportsCheck = [ "timing_asgi" ];
+
+  # Tests rely on asynctest which is not supported by python 3.11
+  doCheck = false;
```

Comment:
Could this be `pythonOlder "3.12"`?

Reply:
Unfortunately no, because `ray` is not available for Python 3.12.
https://github.com/ray-project/ray/issues/40211